src/app/space/settings/collaborators/add-collaborators-dialog/add-collaborators-dialog.component.ts

```diff
@@ -22,66 +22,48 @@ export class AddCollaboratorsDialogComponent implements OnInit {
 
   @Output() onAdded = new EventEmitter<User[]>();
 
-  @ViewChild('dropdown') dropdown: MultiselectDropdown;
-
-  public dropdownOptions: IMultiSelectOption[] = [];
-  public dropdownModel: User[];
-  public dropdownSettings: IMultiSelectSettings;
+  collaborators: User[] = [];
+  selectedCollaborators: User[];
+  loading: Boolean = false;
+  searchTerm = new Subject<string>();
 
   constructor(
     private userService: UserService,
     private collaboratorService: CollaboratorService
   ) { }
 
   ngOnInit() {
-    this.dropdownSettings = {
-      pullRight: false,
-      enableSearch: true,
-      checkedStyle: 'checkboxes',
-      buttonClasses: 'btn btn-default',
-      selectionLimit: 0,
-      closeOnSelect: false,
-      showCheckAll: false,
-      showUncheckAll: false,
-      dynamicTitleMaxItems: 3,
-      maxHeight: '300px'
-    };
+    this.searchTerm.pipe(
+      tap(() => console.log('Something')),
+      distinctUntilChanged(),
+      debounceTime(200),
+      switchMap(term => this.userService.getUsersBySearchString(term))
+    ).subscribe(users => {
+      this.collaborators = users;
+      this.loading = false;
```

Comment:
`this.loading` is never set to `true`

Reply:
this.loading was supposed to be set true in the tap() but by mistake I left console.log statement after testing something there. Updated.

Reply:
I think the order of your pipe should be `debounceTime` => `distinctUntilChanged` => `tap`.
* `debounceTime` first to ensure efficient event processing by waiting 200ms before doing any work.
* `distinctUntilChanged` only after debounced so that comparisons to previous value is minimized.
* `tap` only after we know we're about to search. So that we aren't showing `loading` while the user simply types. Otherwise loading can go on forever if they return back to their previous value due to `distinctUntilChanged`.

```
debounceTime(200),
distinctUntilChanged(),
tap(() => this.loading = true),
```